servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -276,48 +275,53 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
     if (lastInode != null) {
       if (lastInode.isDirectory() && options.isDirectory()) {
         if (!lastInode.isPersisted() && options.isPersisted()) {
-          lastInode.setPersisted(true);
-          traversalResult.getPersisted().add(lastInode);
-          String ufsPath = mMountTable.resolve(getPath(lastInode)).toString();
-          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
-          ufs.mkdirs(ufsPath, false);
+          // The final path component already exists and is not persisted, so it should be added
+          // to the non-persisted Inodes of traversalResult.
+          traversalResult.getNonPersisted().add(lastInode);
```

Comment:
It's intentional. `toPersistInodes` are collections of all inodes that should be persisted, including both those existing and newly created. But `traversalResult.getNonPersisted()`  only contains existing inodes that are persisted during path creation, these inode ids should be journaled in `PersistDirectoryEntry`.